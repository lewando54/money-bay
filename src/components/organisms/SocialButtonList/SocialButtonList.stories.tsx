/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import SocialButtonList from './SocialButtonList'
import { create } from 'react-test-renderer'

const meta = {
    component: SocialButtonList,
    argTypes: {
        socials: { control: 'array' }
    }
} satisfies Meta<typeof SocialButtonList>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        socials: [
            { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
            { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
            { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
        ],
        testId: 'sociallisttest'
    }
}