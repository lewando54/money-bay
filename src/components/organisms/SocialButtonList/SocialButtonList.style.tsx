import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div: {
        display: 'flex',
        width: '100%',
        justifyContent: 'center',
        gap: 10,
        flexDirection: 'row'
    }
})