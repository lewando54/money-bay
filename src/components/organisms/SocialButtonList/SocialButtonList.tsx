import React from 'react'
import { View } from 'react-native'

import { SocialButton } from '@atoms'
import { TiconProp } from 'src/components/atoms/SocialButton/SocialButton'

import socialButtonListStyle from './SocialButtonList.style'

export type TSocialsArray = Array<{ id: number, color: string, icon: TiconProp, href: URL['href'], testId?: string }>
interface ISocialButtonList {
  socials: TSocialsArray
  testId?: string
}

function SocialButtonList ({ socials, testId }: ISocialButtonList): React.ReactElement {
    const socialItems = socials.map(social =>
        <SocialButton size={28} key={social.id} color={social.color} href={social.href} icon={social.icon} testId={testId + social.id.toString()}/>
    )

    return (
        <View testID={testId} style={socialButtonListStyle.div}>
            {socialItems}
        </View>
    )
}

export default SocialButtonList
