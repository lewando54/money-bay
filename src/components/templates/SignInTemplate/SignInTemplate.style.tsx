import { StyleSheet } from 'react-native'
import globalThemeStyle, { MAIN_DARK } from '../../../styling/GlobalTheme.style'
export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        justifyContent: 'center',
        paddingHorizontal: 20,
        gap: 25
    },
    heading: {
        textAlign: 'center',
        fontFamily: 'MulishBold',
        fontSize: 44
    },
    rememberAndPasswordContainer: {
        flexDirection: 'row'
    },
    rememberContainer: {
        flexDirection: 'row',
        gap: 14
    },
    rememberText: {
        ...globalThemeStyle.text_Regular,
        color: MAIN_DARK,
        fontSize: 16
    },
    registerTextContainer: {
        flexDirection: 'row',
        gap: 5
    }
})
