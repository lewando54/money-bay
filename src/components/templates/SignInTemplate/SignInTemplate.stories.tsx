/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React, { useState } from 'react'
import { Meta,StoryObj } from '@storybook/react-native'
import { TSocialsArray } from 'src/components/organisms/SocialButtonList/SocialButtonList'

import SignInTemplate from './SignInTemplate'

interface IStatefulSignInMockupProps {
    emailValue?: string
    passwordValue?: string
    rememberMeValue?: boolean
    socials?: TSocialsArray
    onLostYourPasswordPressHandle: () => void
    onRegisterNowPressHandle: () => void
    handleSubmit: () => void
}

function StatefulSignIn({
    emailValue,
    passwordValue,
    rememberMeValue,
    socials,
    onLostYourPasswordPressHandle,
    onRegisterNowPressHandle,
    handleSubmit

}: IStatefulSignInMockupProps){
    const [email, setEmail] = useState(emailValue)
    const [password, setPassword] = useState(passwordValue)
    const [rememberMe, setRememberMe] = useState(rememberMeValue)

    return (
        <SignInTemplate
            socials={socials}
            emailValue={email}
            passwordValue={password}
            rememberMeChecked={rememberMe}
            onEmailChange={(newValue) => setEmail(() => newValue)}
            onPasswordChange={(newValue) => setPassword(() => newValue)}
            onRememberMeChange={(newValue) => setRememberMe(() => newValue)}
            onLostYourPasswordPress={onLostYourPasswordPressHandle}
            onRegisterNowPress={onRegisterNowPressHandle}
            onSubmit={handleSubmit}
        />
    )
}

const meta = {
    component: StatefulSignIn,
    argTypes: {
        socials: { control: 'object' },
        emailValue: { control: 'text' },
        passwordValue: { control: 'text' },
        rememberMeValue: { control: 'boolean' },
        onLostYourPasswordPressHandle: {action: 'LostYourPassword Pressed'},
        onRegisterNowPressHandle: {action: 'RegisterNow Pressed'},
        handleSubmit: {action: 'Submitted'},
    }
} satisfies Meta<typeof StatefulSignIn>

export default meta
type Story = StoryObj<typeof StatefulSignIn>

export const Default: Story = {
    args: {
        socials: [
            { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
            { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
            { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
        ],
        emailValue: 'test@test.com',
        passwordValue: 'test',
        rememberMeValue: false
    }
}
