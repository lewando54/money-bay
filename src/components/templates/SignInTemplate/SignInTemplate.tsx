import React from 'react'
import { View } from 'react-native'

import { CustomText } from '@atoms'
import { Button, CheckBoxSquare, Link } from '@atoms'
import { EmailInput, PasswordInput } from '@molecules'
import { SocialButtonList } from '@organisms'
import { TSocialsArray } from 'src/components/organisms/SocialButtonList/SocialButtonList'

import signInTemplateStyle from './SignInTemplate.style'

interface ISignInTemplateProps {
    emailValue?: string
    passwordValue?: string
    rememberMeChecked?: boolean
    onEmailChange?: (newValue: string) => void
    onPasswordChange?: (newValue: string) => void
    onRememberMeChange?: (newValue: boolean) => void
    onLostYourPasswordPress?: () => void
    onRegisterNowPress?: () => void
    socials?: TSocialsArray
    onSubmit?: () => void
    testID?: string
}

export default function SignInTemplate({
    emailValue,
    passwordValue,
    rememberMeChecked,
    onEmailChange,
    onPasswordChange,
    onRememberMeChange,
    onLostYourPasswordPress,
    onRegisterNowPress,
    socials,
    onSubmit,
    testID
}: ISignInTemplateProps){
    return (
        <View style={signInTemplateStyle.mainContainer} testID={testID}>
            <CustomText style={signInTemplateStyle.heading}>Welcome to MoneyBay!</CustomText>
            <EmailInput value={emailValue} onChange={onEmailChange}/>
            <PasswordInput value={passwordValue} onChange={onPasswordChange}/>
            <View style={signInTemplateStyle.rememberAndPasswordContainer}>
                <View style={signInTemplateStyle.rememberContainer}>
                    <CheckBoxSquare checked={rememberMeChecked} onPress={onRememberMeChange}/>
                    <CustomText style={signInTemplateStyle.rememberText}>Remember me</CustomText>
                </View>
                <Link style={{marginLeft: 'auto'}} onPress={onLostYourPasswordPress}>Lost your password?</Link>
            </View>
            <Button onClick={onSubmit}>Sign in</Button>
            <View style={signInTemplateStyle.registerTextContainer}>
                <CustomText style={signInTemplateStyle.rememberText}>No account?</CustomText>
                <Link onPress={onRegisterNowPress}>Register now</Link>
            </View>
            <SocialButtonList socials={socials}/>
        </View>
    )
}
