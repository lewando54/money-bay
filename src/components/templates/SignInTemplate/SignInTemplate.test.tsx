import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import SignInTemplate from './SignInTemplate'
import { TSocialsArray } from '../../organisms/SocialButtonList/SocialButtonList'
describe('SignInTemplate', () => {
    const socials: TSocialsArray = [
        { id: 0, color: '#3B5999', icon: 'facebook', href: 'https://google.pl/' },
        { id: 1, color: '#55ACEE', icon: 'twitter', href: 'https://google.pl/' },
        { id: 2, color: '#DD4B39', icon: 'google', href: 'https://google.pl/' }
    ]

    it('should render correctly', () => {
        const tree = renderer.create(<SignInTemplate socials={socials}/>)
        expect(tree).toMatchSnapshot()
    })
})
