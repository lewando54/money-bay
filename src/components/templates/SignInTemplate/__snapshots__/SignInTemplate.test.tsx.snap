// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInTemplate should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "gap": 25,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
        },
        {
          "fontFamily": "MulishBold",
          "fontSize": 44,
          "textAlign": "center",
        },
      ]
    }
  >
    Welcome to MoneyBay!
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      placeholderTextColor="#868698"
      secureTextEntry={false}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        [
          {
            "alignItems": "center",
            "borderColor": "white",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 14,
          },
          undefined,
          false,
        ],
      ]
    }
    testID="universal-container"
  >
    <TextInput
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      placeholderTextColor="#868698"
      secureTextEntry={true}
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "marginRight": 10,
            "maxWidth": "93%",
            "width": "100%",
          },
          undefined,
          {
            "maxWidth": "100%",
          },
        ]
      }
      testID="passwordtest"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={10}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "display": "none",
        }
      }
      testID="input-icon"
    >
      <Text />
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 14,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          animatedStyle={
            {
              "value": {
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "borderColor": "rgba(134, 134, 152, 1)",
              },
            }
          }
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "borderColor": "rgba(134, 134, 152, 1)",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 22,
              "justifyContent": "center",
              "width": 22,
            }
          }
        >
          <View
            animatedStyle={
              {
                "value": {
                  "opacity": 0,
                  "transform": [
                    {
                      "rotateZ": "180deg",
                    },
                  ],
                },
              }
            }
            collapsable={false}
            style={
              {
                "opacity": 0,
                "transform": [
                  {
                    "rotateZ": "180deg",
                  },
                ],
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
          ]
        }
      >
        Remember me
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "#FF5887",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            {
              "marginLeft": "auto",
            },
          ],
        ]
      }
    >
      Lost your password?
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4288142079,
            4279654301,
          ],
          "endPoint": [
            1,
            0.5,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0.5,
          ],
        }
      }
      style={
        [
          {
            "borderRadius": 10,
            "padding": 15,
          },
          false,
          false,
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ],
          ]
        }
      >
        Sign in
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
        ]
      }
    >
      No account?
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "#FF5887",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            undefined,
          ],
        ]
      }
    >
      Register now
    </Text>
  </View>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "gap": 10,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined0"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#3B5999",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined1"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#55ACEE",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="undefined2"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
              "padding": 8,
            },
            {
              "backgroundColor": "#DD4B39",
              "borderRadius": 49,
              "height": 49,
              "width": 49,
            },
          ]
        }
        testID="pressable-social-button"
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
